### PR TITLE
x1032 Add user full name to Release Recipient

### PR DESCRIPTION
**Pull request by sabrine33**

### CODE REVIEW

File: src/components/entityManager/EntityManager.tsx [hunk 1]
```diff
@@ -66,8 +73,31 @@ type EntityManagerProps<E> = {
...
+  /**
+   * Extra property of the entity to display in the table
+   */
+  extraDisplayColumnName?: ExtraEntityColumn<E>;
+};
+
+type BasicReleaseRecipient = {
+  userFullName: string;
+  username: string;
+  enabled: boolean;
```
Comment (seenanair) [inline src/components/entityManager/EntityManager.tsx:86]: Why this need to be redefined as we have already a graphql fragment - ReleaseRecipientFieldsFragment available?

File: src/pages/Configuration.tsx [hunk 2]
```diff
@@ -347,7 +347,7 @@ export default function Configuration({ configuration }: ConfigurationParams) {
           also available on the <StyledLink to={'/sgp'}>SGP Management</StyledLink> page as "Work Requester"
         </p>
         <EntityManager
-          initialEntities={configuration.releaseRecipients}
+          initialEntities={convertToBasicReleaseRecipient(configuration.releaseRecipients)}
```
Comment (seenanair) [inline src/pages/Configuration.tsx:350]: Why we need these conversions?

**Review by seenanair — COMMENTED** [review]
I have seen the following issues 
- The layout of columns are not correct
- The recipient names are not saved or updated

**Comment by seenanair** [comment]
Screenshot 2023-09-06 at 15 16 55
Screenshot 2023-09-06 at 15 16 34

File: src/components/entityManager/EntityManager.tsx [hunk 3]
```diff
@@ -66,8 +73,31 @@ type EntityManagerProps<E> = {
...
+): BasicReleaseRecipient[] => {
+  if (!Array.isArray(recipients)) {
+    recipients = [recipients];
+  }
+  return recipients.map(({ userFullName, username, enabled }) => ({
+    userFullName: userFullName || '',
+    username,
+    enabled
+  }));
+};
```
[the remark this answers is not in the document]
Comment (sabrine33) [reply src/components/entityManager/EntityManager.tsx:100]: As the userFullName field is optional, the conversion is only to set it to empty if undefined

[the remark this answers is not in the document]
Comment (seenanair) [reply src/components/entityManager/EntityManager.tsx:100]: Ok, is it for displaying purpose? If so, is it not possible to check it on the fly and display an empty string?

File: src/pages/Configuration.tsx [hunk 4]
```diff
@@ -347,33 +347,49 @@ export default function Configuration({ configuration }: ConfigurationParams) {
...
           extraDisplayColumnName={{
-            label: 'User Full Name',
-            value: 'userFullName',
-            onChange: (username, userFullName) => {
-              return stanCore
-                .UpdateReleaseRecipientFullName({ username, userFullName })
-                .then((res) => convertToBasicReleaseRecipient(res.updateReleaseRecipientFullName)[0]);
+            label: 'Full Name',
+            value: 'fullName',
+            extraFieldPlaceholder: 'Enter User Full Name',
```
Comment (seenanair) [inline src/pages/Configuration.tsx:384]: Probably you can change this as well to  'Enter full name' to match column header

File: src/components/entityManager/EntityManager.tsx [hunk 5]
```diff
@@ -290,7 +272,7 @@ export default function EntityManager<E extends Record<string, EntityValueType>>
           <TableCell colSpan={2}>
             <Input
```
Comment (seenanair) [inline src/components/entityManager/EntityManager.tsx:273]: Can we have the class style for the full name <Input> to be as no border  (or a very subtle one) as in normal condition and when we start editing (that means when it gets focus), the border can appear? This way the field will look  more similar to other, I think

**Review by seenanair — COMMENTED** [review]
Looking good 👍

**Review by seenanair — APPROVED** [review]

**Pull request merged**